providers/shared/components/component.ftl

```diff
@@ -36,6 +39,7 @@
 
 [#assign DB_COMPONENT_TYPE = "db" ]
 [#assign DB_LEGACY_COMPONENT_TYPE = "rds" ]
+[#assign legacyTypeMapping += { DB_LEGACY_COMPONENT_TYPE : DB_COMPONENT_TYPE } ]
```

Comment:
I thought you had this on the Id for the DB Component? Any reason this moved?

Reply:
I think I like that its part of where you set the component structure. Makes it managed from a central point then

Reply:
Totally agree but unfortunately we need to know its a legacy type to work out its canonical type to then know which component to lazy load. So at the point of analysis, nothing from the component macro has been loaded.

As this file is where all the types live, it felt like the next best spot.